other/coding/src/main/java/org/example/coding/moneytransfer/MoneyService.java

```diff
@@ -0,0 +1,60 @@
+package org.example.coding.moneytransfer;
+
+import org.example.coding.moneytransfer.ops.AccountOperations;
+import org.example.coding.moneytransfer.model.Account;
+
+import java.util.Optional;
+
+public class MoneyService {
+
+    private final AccountOperations accountOperations;
+
+    public MoneyService(AccountOperations accountOperations) {
+        this.accountOperations = accountOperations;
+    }
+
+    public boolean addMoney(int accountId, int amount) {
+        Optional<Account> accountOpt = accountOperations.getAccount(accountId);
+        if (accountOpt.isEmpty()) {
+            return false;
+        }
+
+        accountOpt.get().addToBalance(amount);
+
+        return true;
+    }
+
+    public boolean extractMoney(int accountId, int amount) {
+        Optional<Account> accountOpt = accountOperations.getAccount(accountId);
+        if (accountOpt.isEmpty()) {
+            return false;
+        }
+
+        return accountOpt.get().extractFromBalance(amount);
+    }
+
+    public boolean transferMoney(int fromAccountId, int toAccountId, int amount) {
+        Optional<Account> accountFromOpt = accountOperations.getAccount(fromAccountId);
+        if (accountFromOpt.isEmpty()) {
+            return false;
+        }
+        Optional<Account> accountToOpt = accountOperations.getAccount(toAccountId);
+        if (accountToOpt.isEmpty()) {
+            return false;
+        }
+
+        Account accountFrom = accountFromOpt.get();
+        Account accountTo = accountToOpt.get();
+
+        if (accountFrom.extractFromBalance(amount)) {
```

Comment:
This `ifs` looks a bit weird to me. `addToBalance` never returns false, so probably could be simplified. Overall i'd rather throw exceptions everywhere, than return `true/false`, i.e. `throw AccountNotFound` or `throw OperationFailed` or something of the kind.